---
id: __ruleName__
description: __description__
category: __category__
severity: __severity__
---

# `__ruleName__`

<!-- prettier-ignore-start -->
__description__
<!-- prettier-ignore-end -->

❌ Examples of **incorrect** code for this rule

```html
<todo>Write incorrect codes</todo>
```

✅ Examples of **correct** code for this rule

```html
<todo>Write correct codes</todo>
```
