---
title: 'TODO: Title'
id: '{{ name }}'
category: 'TODO category'
---

# TODO: Title

TODO: Write a description

## Rule Details

üëé Examples of **incorrect** code for this rule

```html
<todo>Write incorrect codes</todo>
```

üëç Examples of **correct** code for this rule

```html
<todo>Write correct codes</todo>
```

### Interface

- Type: `string`
- Deafult Value: `"__DEFAULT_MAIN_VALUE__"`

### Options

TODO: Write a description

#### Interface

| Property | Type       | Optional | Default Value | Description               |
| -------- | ---------- | -------- | ------------- | ------------------------- |
| `foo`    | `string`   | ‚úî        | `undefined`   | TODO: Write a description |
| `bar`    | `number[]` | ‚úî        | `undefined`   | TODO: Write a description |

### Default severity

TODO: Choose `error` or `warning`
