---
title: å±æ€§é–“ã®ã‚¹ãƒšãƒ¼ã‚¹
id: attr-spacing
fixable: true
cateogry: style
---

# å±æ€§é–“ã®ã‚¹ãƒšãƒ¼ã‚¹

è¦ç´ é–“ã®ã‚¹ãƒšãƒ¼ã‚¹ãƒ»æ”¹è¡Œã®æœ‰ç„¡ã‚„æ•°ã«å¯¾ã—ã¦è­¦å‘Šã—ã¾ã™ã€‚

**ğŸ”§ Fixable**

## Rule Details

ğŸ‘ Examples of **incorrect** code for this rule

<!-- prettier-ignore-start -->
```html
<img src="path/to"src="path/to2">
```
<!-- prettier-ignore-end -->

ğŸ‘ Examples of **correct** code for this rule

<!-- prettier-ignore-start -->
```html
<img src="path/to" src="path/to2">
```
<!-- prettier-ignore-end -->

### Setting value

Type: `boolean`

### Options

#### `lineBreak`

Type: `"either" | "always" | "never"`

| value      | default | description                    |
| ---------- | ------- | ------------------------------ |
| `"either"` | âœ“       | æ”¹è¡Œã«é–¢ã—ã¦è­¦å‘Šã‚’ã—ã¾ã›ã‚“ã€‚   |
| `"always"` |         | æ”¹è¡Œã‚’ã—ã¦ã„ãªã„ã¨è­¦å‘Šã—ã¾ã™ã€‚ |
| `"never"`  |         | æ”¹è¡Œã‚’ã—ã¦ã„ã‚‹ã¨è­¦å‘Šã—ã¾ã™ã€‚   |

#### `width`

Type: `number | false`

| value    | default | description                                              |
| -------- | ------- | -------------------------------------------------------- |
| [number] | âœ“ `1`   | ã‚¹ãƒšãƒ¼ã‚¹ã®å¹…ãŒè¨­å®šã—ãŸæ•°å€¤ã«ãªã£ã¦ã„ãªã‘ã‚Œã°è­¦å‘Šã—ã¾ã™ã€‚ |
| `false`  |         | ã‚¹ãƒšãƒ¼ã‚¹ã®å¹…ã«é–¢ã—ã¦è­¦å‘Šã‚’ã—ã¾ã›ã‚“ã€‚                     |

### Default notification severity

`warning`
